instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/LogbackAppenderApplicationListener.java

```diff
@@ -63,12 +63,25 @@ public void onApplicationEvent(ApplicationEvent event) {
       if (existingOpenTelemetryAppender.isPresent()) {
         reInitializeOpenTelemetryAppender(
             existingOpenTelemetryAppender, applicationEnvironmentPreparedEvent);
-      } else {
+      } else if (isLogbackAppenderAddable(applicationEnvironmentPreparedEvent)) {
         addOpenTelemetryAppender(applicationEnvironmentPreparedEvent);
       }
     }
   }
 
+  private static boolean isLogbackAppenderAddable(
```

Comment:
why not use `@Conditional(SdkEnabled.class)` [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/48924cb9697ae82cc76f78d8e7f0a95955fb80bd/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/OpenTelemetryAppenderAutoConfiguration.java#L17) instead?

Reply:
The application listener is executed very early at the application startup. `@Conditional(SdkEnabled.class)` can't work. The property has to be retrieved from an application event.

Reply:
Could you add the underlying `@ConditionalOnProperty` to the configuration?

Reply:
> why not use `@Conditional(SdkEnabled.class)` [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/48924cb9697ae82cc76f78d8e7f0a95955fb80bd/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/OpenTelemetryAppenderAutoConfiguration.java#L17) instead?

I did not read your question correctly . The link you gave is related to application listeners allowing to inject an OpenTelemetry instance into the OpenTelemetry appenders. This PR has a different goal. It aims to be able to disable the automatic addition of the Logback OpenTelemetry appender. You have already added `@Conditional(SdkEnabled.class)` to  the autoconfiguration creating the listeners allowing the injection of the OpenTelemetry instance into the OpenTelemetry appenders in #10602: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b0bbbd17384e6c8eb7bd271c0aa4aa37b5b1c991/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/OpenTelemetryAppenderAutoConfiguration.java#L21

Reply:
Right - next try: check for sdk disabled at the start of the method to make it very clear:

```
public void onApplicationEvent(ApplicationEvent event) {
    if (event instanceof ApplicationEnvironmentPreparedEvent // Event for which
    // org.springframework.boot.context.logging.LoggingApplicationListener
    // initializes logging
    ) {
      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent =
          (ApplicationEnvironmentPreparedEvent) event;
      Boolean otelSdkDisableProperty =
          evaluateBooleanProperty(applicationEnvironmentPreparedEvent, "otel.sdk.disabled");
      if (otelSdkDisableProperty != null && otelSdkDisableProperty) {
        return;
      }
      
      Optional<OpenTelemetryAppender> existingOpenTelemetryAppender = findOpenTelemetryAppender();
      if (existingOpenTelemetryAppender.isPresent()) {
        reInitializeOpenTelemetryAppender(
            existingOpenTelemetryAppender, applicationEnvironmentPreparedEvent);
      } else if (isLogbackAppenderAddable(applicationEnvironmentPreparedEvent)) {
        addOpenTelemetryAppender(applicationEnvironmentPreparedEvent);
      }
    }
  }
```

Reply:
> Right - next try: check for sdk disabled at the start of the method to make it very clear:
> 
> ```java
> public void onApplicationEvent(ApplicationEvent event) {
>     if (event instanceof ApplicationEnvironmentPreparedEvent // Event for which
>     // org.springframework.boot.context.logging.LoggingApplicationListener
>     // initializes logging
>     ) {
>       ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent =
>           (ApplicationEnvironmentPreparedEvent) event;
>       Boolean otelSdkDisableProperty =
>           evaluateBooleanProperty(applicationEnvironmentPreparedEvent, "otel.sdk.disabled");
>       if (otelSdkDisableProperty != null && otelSdkDisableProperty) {
>         return;
>       }
>       
>       Optional<OpenTelemetryAppender> existingOpenTelemetryAppender = findOpenTelemetryAppender();
>       if (existingOpenTelemetryAppender.isPresent()) {
>         reInitializeOpenTelemetryAppender(
>             existingOpenTelemetryAppender, applicationEnvironmentPreparedEvent);
>       } else if (isLogbackAppenderAddable(applicationEnvironmentPreparedEvent)) {
>         addOpenTelemetryAppender(applicationEnvironmentPreparedEvent);
>       }
>     }
>   }
> ```

It would remove the ability to reconfigure an OpenTelemetry Logback appender defined in an XML file from system properties.

Imagine a user having configured the capture of code attributes from an OTel Logback appender in an XML file. The user has packaged his application and is using it  in production. The capture of code attributes is expensive and can cause performance penalty. It would not be possible to disable the capture of log attributes without repacking the application and redeliver it in production.